server/tests/backends/sql_translator/test_sql_postgres_pypika_translator_steps.py

```diff
@@ -0,0 +1,172 @@
+    docker_container.stop()
+
+
+# Update this method to use snowflake connection
+def get_connection():
+    con_params = {
+        'host': '127.0.0.1',
+        'user': 'ubuntu',
+        'password': 'ilovetoucan',
+        'port': 5432,
+        'database': 'pg_db',
+    }
+    conn = psycopg2.connect(**con_params)
+    return conn
+
+
+@pytest.fixture
+def get_engine():
+    host = '127.0.0.1'
+    user = 'ubuntu'
+    password = 'ilovetoucan'
+    port = 5432
+    database = 'pg_db'
+    engine = create_engine(f'postgresql://{user}:{password}@{host}:{port}/{database}')
+    return engine
+
+
+def execute(connection, query: str, meta: bool = True) -> Optional[pd.DataFrame]:
+    with connection.cursor() as cursor:
+        cursor.execute(query)
+        if meta:
+            df = pd.DataFrame(cursor.fetchall())
+            field_names = [i[0] for i in cursor.description]
+            df.columns = field_names
+            return df
+        else:
+            return None
+
+
+def standardized_columns(df: pd.DataFrame):
+    df.columns = [c.replace('-', '_') for c in df.columns]
+
+
+# Translation from Pipeline json to SQL query
+@pytest.mark.parametrize('case_id, case_spec_file_path', test_cases)
+def test_sql_translator_pipeline(case_id, case_spec_file_path, get_engine):
+    spec = get_spec_from_json_fixture(case_id, case_spec_file_path)
+    test_table_name = case_id.replace("/", "")
+
+    # Drop created table
+    execute(get_connection(), f'DROP TABLE IF EXISTS {test_table_name}', False)
+
+    # inserting the data in Postgres
+    # Take data in fixture file, set in pandas, create table and insert
+    data_to_insert = pd.read_json(json.dumps(spec['input']), orient='table')
+    standardized_columns(data_to_insert)
+    data_to_insert.to_sql(
+        name=case_id.replace('/', ''), con=get_engine, index=False, if_exists='replace', chunksize=1
+    )
+
+    if 'other_inputs' in spec:
+        for input in spec['other_inputs']:
+            data_other_insert = pd.read_json(
+                json.dumps(spec['other_inputs'][input]), orient='table'
+            )
+            standardized_columns(data_other_insert)
+            data_other_insert.to_sql(
+                name=input,
+                con=get_engine,
+                index=False,
+                if_exists='replace',
+            )
+
+    steps = spec['step']['pipeline']
+    steps.insert(0, {'name': 'domain', 'domain': test_table_name})
+    pipeline = PipelineWithVariables(steps=steps)
+
+    # Convert Pipeline object to Postgres Query
+    query = translate_pipeline(
+        sql_dialect=SQLDialect.POSTGRESQL,
+        pipeline=pipeline,
+        tables_columns={test_table_name: data_to_insert.columns},
+        db_schema=None,
+    )
+    # Execute request generated from Pipeline in Postgres and get the result
+    result: pd.DataFrame = execute(get_connection(), query)
+
+    # Drop created table
+    execute(get_connection(), f'DROP TABLE {test_table_name}', False)
```

Comment:
Is this needed ? The container is removed after the tests anyway. No big deal but could be an easy speedup

Reply:
agreed.